superset-frontend/src/dashboard/components/DeleteComponentButton.tsx

```diff
@@ -16,25 +16,21 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { MouseEventHandler } from 'react';
 import Icons from 'src/components/Icons';
 import IconButton from './IconButton';
 
-const propTypes = {
-  onDelete: PropTypes.func.isRequired,
+type DeleteComponentButtonProps = {
+  onDelete: MouseEventHandler<HTMLDivElement>;
```

Comment:
Is this the function type? Could it not be a () => {} like most functions?

Reply:
Yes, that is a function type. @AAfghahi 
Of course it can `() => {}`.
I recommend keeping it for the type safety since it's used on the `onDelete` event.
Do you still want me to change it?
![image](https://user-images.githubusercontent.com/37535505/151516729-7f882af0-b104-45f4-b582-75c7538b7644.png)

Reply:
Ok, if it is the same then that is fine. I just wanted to amek sure.